src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -26,16 +26,29 @@ object WdlExpression {
 
   val unaryOperators = Set("LogicalNot", "UnaryPlus", "UnaryNegation")
 
-  def evaluate(ast: AstNode, lookup: ScopedLookupFunction, functions: WdlFunctions): Try[WdlValue] = {
+  def interpolate(str: String, lookup: ScopedLookupFunction): String = {
+    val regex = "\\$\\{([a-z]+)\\}".r
+    var interpolated = str
```

Comment:
Prefer immutability. You could do this with a fold where each iteration it's being passed the previous iteration